tools/codegen/src/generator/languages/c/mod.rs

```diff
@@ -33,6 +33,11 @@ impl Generator {
         w!(o, "#define _CPP_BEGIN extern \"C\" {{                     ");
         w!(o, "#define _CPP_END }}                                    ");
         w!(o, "_CPP_BEGIN                                             ");
+        w!(o, "                                                       ");
+        w!(o, "#ifndef MOLECULE_API_DECORATOR                         ");
```

Comment:
If `#ifdef __cplusplus`return false, the `MOLECULE_API_DECORATOR` will not be defined (in this file).

In current version, the code works.
Because the `MOLECULE_API_DECORATOR` was defined in `molecule_reader.h`, and it wasn't `undef` (side effects?).

I will update this in another PR.